vars/buildDotNetCore.groovy

```diff
@@ -34,10 +34,15 @@ def call(Map config=[:]) {
       stage('Build test image') {
         build.buildTestImage(DOCKER_REGISTRY_CREDENTIALS_ID, DOCKER_REGISTRY, repoName, BUILD_NUMBER, tag)
       }
+
       stage('Run tests') {
         build.runTests(repoName, repoName, BUILD_NUMBER, tag)
       }
 
+      stage('SonarCloud analysis') {
+        test.analyseDotNetCode(test.buildCodeAnalysisDotNetParams(repoName, BRANCH_NAME, pr))        
```

Comment:
It looks a little odd passing the output of the method a class to another method of the same class from an external method. 
Could the `repoName`, `BRANCH_NAME`, `and pr` parameters be pulled up to the `analyseDotNetCode` method so only one method call needs tobe exposed?

Reply:
Yes agreed that would be better.  Must confess I just copied the pattern we have in the Node.js pipeline which in turn mirrors what we originally had with SonarQube.  Now the library has been restructured it does look a bit odd.

I've also updated the Node.js pipeline to reflect this suggestion.

You can see that the changes still have the intended behaviour with this test PR.
https://sonarcloud.io/dashboard?id=ffc-demo-payment-service-core&pullRequest=79